### ISSUE TITLE
DAO fetch methods should use GeneratorStrategy::getJavaMemberName with Mode.DAO instead of getJavaClassName

### ISSUE THREAD

**Issue by dwikle** [jOOQ/jOOQ#7668]
### Expected behavior and actual behavior:

I am attempting to use a custom generator strategy to add a prefix to jooq-generated classes, as demonstrated in `JPrefixGeneratorStrategy`.  I was surprised that the generated DAO methods such as `fetchOneBy<Column>` also had the prefix inserted before the column name. 

In the sample below I was expecting the fetch method to be unaffected by the Java class name, so that it would be `fetchOneById`.  However it is generated as `fetchOneByJId`. Is that the intended behavior?
  
### Steps to reproduce the problem (include example code if possible):

Given a table like this:
```
CREATE TABLE example.user (
  id INT PRIMARY KEY
);
```

Generate code using the following:
```
public class JPrefixGeneratorStrategy extends DefaultGeneratorStrategy {
    @Override
    public String getJavaClassName(final Definition definition, final Mode mode) {
        return 'J' + super.getJavaClassName(definition, mode);
    }
}
```

Note the naming of the fetch methods in the resulting DAO for the table `JUserDao`:
```
public com.bamnetworks.cce.database.jooq.tables.pojos.JUser fetchOneByJId(Integer value) {
    return fetchOne(JUser.USER.ID, value);
}
```

I'm on an older version of jOOQ but from reading the code it looks like the most recent version would behave in the same manner.

### Versions:

- jOOQ: 3.7.4
- Java: 1.8
- Database: Postgres

**Comment by lukaseder** [comment]
Thank you very much for your report.

Historically, the `getJavaClassName()` method was abused for any thing that by default generated a `PascalCase` name. Which is silly, of course. There should be new, dedicated methods per use-case, which might just happen to work the same way as `getJavaClassName()` by default.

**Comment by lukaseder** [comment]
Fixing this will probably constitute an incompatible change, as people already rely on this behaviour. Nevertheless, it's worth cleaning these things up.

**Comment by lukaseder** [comment]
Related:

- #13028

**Comment by lukaseder** [comment]
Doing this now for 3.20. There should also be matcher strategy support via `<daoMember/>`:

https://www.jooq.org/doc/latest/manual/code-generation/codegen-matcherstrategy/codegen-matcher-fields/

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.20

**Comment by lukaseder** [comment]
I think that's a more generic problem, which I'll solve some other day.
